Replace custom win32 _ssprintf functions (`werr_ssprintf`, `wo_ssprintf`, `hr_ssprintf`) with safer alternatives

sukibaby opened the pull request:
#827 won't build on Windows until these `.c_str`'s are also applied.

I know in #834 some of these changes were deemed not necessary, but it will not build on MSVC without these additional `.c_str()`'s.

sukibaby committed:
Only use string literals when logging.

The main change here is to bring string construction out of the log itself, and change Windows specific functions werr_ssprintf and hr_ssprintf to not have an internal ssprintf call. Now we construct the string separately, and then pass the constructed string to the Log or Throw call for added safety. Additionally some braces were added where they were missing. Additionally note in DSoundHelpers I changed which variable is being statically cast, to make it more clear to the reader that this is never intended to be a negative number, since previously the DWORD was cast to int.

aeubanks commented:
these `werr_ssprintf` usages are annoying. we really shouldn't allow non-string literal uses of `LOG->Trace/Warn/...` so that the compiler can warn on invalid printf usages.

I'd like

```
LOG->Trace(werr_ssprintf(GetLastError(), "foo %s", s));
```

to instead be

```
LOG->Trace("foo %s (%s)", s, werr(GetLastError()));
```

where (see `src/archutils/Win32/ErrorStrings.cpp`)

```
RString werr(int err) {
  	char buf[1024] = "";
	FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
		0, err, 0, buf, sizeof(buf), nullptr);

	// Why is FormatMessage returning text ending with \r\n? (who? -aj)
	// Perhaps it's because you're on Windows, where newlines are \r\n. -aj
	RString text = buf;
	Replace(text, "\n", "" );
	Replace(text, "\r", " " ); // foo\r\nbar -> foo bar
	TrimRight( text ); // "foo\r\n" -> "foo"
        return text;
}
```

ditto with `hr_ssprintf`

sukibaby commented:
> these `werr_ssprintf` usages are annoying. we really shouldn't allow non-string literal uses of `LOG->Trace/Warn/...` so that the compiler can warn on invalid printf usages.

agreed. I would be more than happy to just get `werr_ssprintf` gone. Maybe it made more sense at the time with c++98 limitations, idk though, mostly seems like some hacky code to me.

> 
> I'd like
> 
> ```
> LOG->Trace(werr_ssprintf(GetLastError(), "foo %s", s));
> ```
> 
> to instead be
> 
> ```
> LOG->Trace("foo %s (%s)", s, werr(GetLastError()));
> ```
> ditto with `hr_ssprintf`

👍

sukibaby committed:
Threads_Win32: refactor two ASSERT_M's into if statements

This is mainly so we can prevent running any of the logging logic unless the condition for failure is met.

sukibaby committed:
LoadingWindow_Win32: refactor two ASSERT_M's into if statements

This is mainly to avoid constructing a failure string unless there actually is a failure. Note we won't actually reach the return statement due to the FAIL_M, but I wanted to include the returns so the code is proper.

sukibaby committed:
ErrorStrings: refactor two ASSERT_M's into if statements

This is mainly to avoid constructing a failure string unless there actually is a failure. Note we won't actually reach the return statement due to the FAIL_M, but I wanted to include the returns so the code is proper.

sukibaby committed:
WaveOut: replace wo_ssprintf

Squashed commit of the following:
commit bb8f75e4a848e6b7d73c15b33c3872707ae77cd2
Author: sukibaby <163092272+sukibaby@users.noreply.github.com>
Date:   Thu Jun 19 02:01:12 2025 -0700

    WaveOut: replace wo_ssprintf

    Improves safety of the function which retrieves waveOutGetErrorText as well as the call sites of this function.

sukibaby committed:
Merge branch 'beta' into string-win-awfga

sukibaby commented:
I'm not sure what the merge conflicts here are. I can't rebase this due to not having write access, is there any chance someone else can try to resolve the merge conflicts here?

sukibaby commented:
Someone who has write permission to the repository needs to handle resolving this issue, as I can't rebase this myself without closing the PR:

<img width="438" height="83" alt="image" src="https://github.com/user-attachments/assets/2f6257d3-ecf6-4172-beb0-8e8bd6f4dfb4" />

I will mark it ready for review so it can be merged if this is resolved.

sukibaby commented:
Feel free to re-open if someone would like to resolve the error shown above. I imagine this PR has to be remade due to conflicts in the history, or rebased by someone with write permissions.
